futures-test/src/lib.rs

```diff
@@ -49,3 +50,16 @@ pub mod io;
 mod assert_unmoved;
 mod interleave_pending;
 mod track_closed;
+
+/// Enables an `async` test function. The generated future will be run to completion with
+/// [`futures_executor::block_on`](futures_executor::block_on).
+///
+/// ```
+/// #[futures_test::test]
+/// async fn my_test() {
+///     let fut = async { true };
+///     assert!(fut.await);
+/// }
+/// ```
```

Comment:
rustdoc does the equivalent of cargo run, not cargo test, so doctests don't actually do anything in this example. Could you add tests to `futures/tests/`?